packages/gdl-frontend/pages/index.js

```diff
@@ -54,75 +54,96 @@ class IndexPage extends React.Component<Props> {
-      // See https://github.com/zeit/next.js/wiki/Redirecting-in-%60getInitialProps%60
-      const redirectUrl = `/${DEFAULT_LANGUAGE.code}`;
-      if (res) {
-        res.writeHead(302, { Location: redirectUrl });
-        res.end();
+      let category: string;
+      if (asPath.includes('/classroom')) {
+        category = 'Classroom';
+      } else if (asPath.includes('/library')) {
+        category = 'Library';
+      } else if (categoryInCookie && categories.includes(categoryInCookie)) {
+        // Small check to make sure the value in the cookie is something valid
+        // $FlowFixMe: We know this is a valid category :/
+        category = categoryInCookie;
       } else {
-        Router.push(redirectUrl);
+        // Default to Library
+        category = categories.includes('Library') ? 'Library' : categories[0];
       }
-      return {};
-    }
-    const categories = categoriesRes.data.categories;
-    const categoryInCookie = getBookCategory(req);
 
-    let category: string;
-    if (asPath.includes('/classroom')) {
-      category = 'Classroom';
-    } else if (asPath.includes('/library')) {
-      category = 'Library';
-    } else if (categoryInCookie && categories.includes(categoryInCookie)) {
-      // Small check to make sure the value in the cookie is something valid
+      const booksAndFeatured: {
+        data: BooksAndFeatured
+      } = await apolloClient.query({
+        query: BOOKS_AND_FEATURED_QUERY,
+        variables: {
+          category,
+          language: languageCode,
+          pageSize: AMOUNT_OF_BOOKS_PER_LEVEL
+        }
+      });
+
       // $FlowFixMe: We know this is a valid category :/
-      category = categoryInCookie;
-    } else {
-      // Default to Library
-      category = categories.includes('Library') ? 'Library' : categories[0];
-    }
+      setBookLanguageAndCategory(languageCode, category, res);
+
+      const {
+        data: { featuredContent, ...bookSummaries }
+      } = booksAndFeatured;
 
-    const booksAndFeatured: {
-      data: BooksAndFeatured
-    } = await apolloClient.query({
-      query: BOOKS_AND_FEATURED_QUERY,
-      variables: {
+      return {
         category,
-        language: languageCode,
-        pageSize: AMOUNT_OF_BOOKS_PER_LEVEL
+        categories,
+        languageCode,
+        // Currently the UI only supports one featured content, not an array
+        featuredContent: featuredContent[0],
+        bookSummaries
+      };
+    } catch (error) {
+      /*
+       * If user request invalid query param to graphql you trigger bad input validation
+       * and recieve 400: Bad Request. The right feedback to the client is a 404 page
+       * and since graphql does not have a better error handling mechanism this is a dirty check.
+       */
+      if (
+        error.graphQLErrors &&
+        error.graphQLErrors.length > 0 &&
+        error.graphQLErrors[0].message === '400: Bad Request'
+      ) {
+        console.log(error.graphQLErrors[0]);
```

Comment:
Trenger vi å ha en `console.log()` her? Hvis så, burde vi ikke ha en for `500` feilen også?

Reply:
Nope, den skulle egt. være der :D

Reply:
Oops, mente at den ikke skulle være der. Sorry!